ruma-events-macros/src/event_enum.rs

```diff
@@ -202,6 +211,113 @@ fn marker_traits(ident: &Ident) -> TokenStream {
     }
 }
 
+fn accessor_methods(ident: &Ident, variants: &[Ident]) -> TokenStream {
+    let origin_server_ts = if has_origin_server_ts(ident) {
```

Comment:
Could you try to remove the repitition of all these if / else blocks (where the else is never really of any interest)? I'm thinking of iterating over `&[("origin_server_ts", has_origin_server_ts), ("room_id", has_room_id), ...]` and having a helper function `fn field_type(field_name: &str) -> TokenStream`.